### DISCUSSION TITLE
Could this integration be used to replace sigenergy AI functionality

### DISCUSSION THREAD

**Discussion by farmboydavy** [TypQxQ/Sigenergy-Home-Assistant-Integration#39] [Q&A]
Hi. I am new to home assistant but work creating and consuming APIs as a software engineer.  I was not sure of the capabilities that are present or could be developed. 

My worries are that the AI services provided by sigenergy will have cost in the future. So wondered if one could build the same capabilities in home assistant and how to do it? For example check future and current electricity prices, weather, consumption patterns and then adjust behaviour accordingly.

**Comment by TypQxQ** [comment]
Yes, you can set the system to change or discharge, use from battery or only from grid. I haven’t tested with EMHASS but should work. 
https://emhass.readthedocs.io/en/latest/intro.html

**Comment by purcell-lab** [comment]
I am currently actively working on integrating EMHASS with SigEnergy, local control via ModBus is great and provides for instant controls..

EMHASS will also allow you to account for not just your solar, battery and EV, but also other key household loads like hot water and HVAC, I have a forecast ready for the V2G component as well, once SigEnergy enable that.

![image](https://github.com/user-attachments/assets/c4fa9e13-b790-44fd-9324-91a5b4cb1caa)
